stor/test.py

```diff
@@ -345,3 +350,38 @@ def teardown_posix_files(self):
     def teardown_project(self):
         self.project_handler.destroy()
         self.project_handler = None
+
+
+class MockFuture():
+    """A class to minimally mock methods for Future objects in ThreadPoolExecutor."""
+    def __init__(self, **kwargs):
+        self.__result = {
+            "success": "complete",
+            "dest": kwargs.get("dest"),
+            "source": kwargs.get("source")
+        }
+
+    def result(self):
+        return self.__result
+
+
+class MockExecutor(Executor):
+    """An Executor class to minimally mock methods for ThreadPoolExecutor."""
+    def __init__(self):
+        self._shutdown = False
+        self._shutdownLock = Lock()
+
+    def submit(self, fn, *args, **kwargs):
+        obj = args[0]
+        if isinstance(obj, dict):
+            dest = obj.get("dest")
+            source = obj.get("source")
+        else:
```

Comment:
I could probably be more specific that this is the upload object...

Reply:
yea I think it would be helpful to add a docstring for these functions describing why you added them i.e. mocking executor.submit or what they do

Reply:
Definitely wanted to add more documentation, which I shall do soon!

Reply:
Added some additional documentation in 2849a72.

Happy to add more if you feel it is not clear enough!